base/parse.jl

```diff
@@ -107,7 +107,10 @@ function tryparse_internal(::Type{T}, s::AbstractString, startpos::Int, endpos::
     end
 
     base = convert(T, base)
-    m::T = div(typemax(T) - base + 1, base)
+    # Special case the common cases of base being 10 or 16 to avoid expensive runtime div
+    m::T = base == 10 ? div(typemax(T) - T(11), T(10)) :
+           base == 16 ? div(typemax(T) - T(16), T(16)) :
```

Comment:
Should these be `- T(9)` and `- T(15)`?

Reply:
Oops, should have ran the tests first heh.